Add PHP 8 as additional requirement

LIQRGV opened the pull request:
Some library that use this package can't get their CI passed since this package restrict them to use PHP 8.
Some example is here
https://github.com/domnikl/statsd-php/pull/77

LIQRGV committed:
Add PHP 8 as additional requirement

Also, we Drop support for old PHP version

PHP with version below 7.3 isn't supported by PHP anymore.
I think it's good idea to drop it too
https://www.php.net/supported-versions.php

LIQRGV commented on appveyor.yml:
```diff
@@ -6,6 +6,15 @@ branches:
 
+environment:
+  matrix:
+    - dependencies: lowest
```
Windows doesn't support version below 8.0, 
so...here we are.

LIQRGV commented:
I saw other PR similar with this on #19. 
But mine has some other changes with appveyor / travis.
Can you take a look on this @webmozart, thank you...

Ocramius commented on .travis.yml:
```diff
@@ -12,9 +12,6 @@ cache:
-    - php: 5.5
     - php: 5.6
     - php: hhvm
     - php: nightly
```
This can now be `8.0`

Ocramius replied:
`allow_failures` should not contain `8.0` anymore

LIQRGV replied:
8.0 added already

Ocramius commented on appveyor.yml:
```diff
@@ -6,6 +6,15 @@ branches:
+      php_ver_target: 7.4
+    - dependencies: highest
+      php_ver_target: 7.4
+
```
We need an entry for `8.0`

Ocramius commented on composer.json:
```diff
@@ -9,7 +9,7 @@
     ],
     "require": {
-        "php": "^5.3.3|^7.0",
+        "php": "^5.3.3|^7.0|^8.0",
```
`5.x` doesn't seem realistic/sensible anymore, but that's up to @webmozart to decide

LIQRGV committed:
Update deps version

LIQRGV committed:
Update globbing rules

According to https://man7.org/linux/man-pages/man7/glob.7.html, now inverse search should using apostrhope instead of caret

LIQRGV committed:
Update php version on travis

Nek- commented on composer.json:
```diff
@@ -9,13 +9,12 @@
     ],
     "require": {
-        "php": "^5.3.3|^7.0",
+        "php": ">=7.2.5",
```
It's not because PHPUnit is not compatible with old codebases that the package should abandon PHP 5.x. Of course, it seems to be useless, but some people may have issues with the package if this change is not done on a feature version besides all the content of this PR is actually suppose to go to a minor release. (speaking for PHP 5.6 users while I'm not part of this team for a long time...)

It's @webmozart to decide.

LIQRGV replied:
How about PHP 5.x user stay on old version ?
I don't think we well have much new feature on the future.
This PR intendend only to support PHP 8 onward

Ocramius replied:
Indeed: old releases will work just fine 👍

chriskapp commented:
`vimeo/psalm` uses also this dependency, so it would be great if we can merge this PR and tag a new release.

```
  Problem 1
    - vimeo/psalm[3.14.0, ..., 3.18.2] require webmozart/glob ^4.1 -> satisfiable by webmozart/glob[4.1.0].
    - webmozart/glob 4.1.0 requires php ^5.3.3|^7.0 -> your php version (8.0.0) does not satisfy that requirement.
```

LIQRGV commented:
]

> `vimeo/psalm` uses also this dependency, so it would be great if we can merge this PR and tag a new release.
> 
> ```
>   Problem 1
>     - vimeo/psalm[3.14.0, ..., 3.18.2] require webmozart/glob ^4.1 -> satisfiable by webmozart/glob[4.1.0].
>     - webmozart/glob 4.1.0 requires php ^5.3.3|^7.0 -> your php version (8.0.0) does not satisfy that requirement.
> ```

Actually, I'm updating this package because I'm using `vimeo/psalm` too @chriskapp 
lol

localheinz commented on composer.json:
```diff
@@ -9,13 +9,12 @@
     ],
     "require": {
-        "php": "^5.3.3|^7.0",
+        "php": ">=7.2.5",
```
How about not using unbounded version constraints here?

Also see https://github.com/symfony/symfony/pull/36876#issuecomment-631351947.

Nek- replied:
https://twitter.com/nicolasgrekas/status/1263023258938548225

https://twitter.com/driesvints/status/1263031553623556099

Ocramius replied:
Got bitten by this yesterday on one of my dep upgrades, and got something ancient because it had `>=7.2` in its constraints.

Let's please not repeat the same mistakes we did with un-bounded dependency ranges in 2012~2013 :-)

LIQRGV replied:
So, you mean we should add ^8.0.0 too?

Ocramius replied:
Correcto 👍

LIQRGV replied:
Done @ocramius

Nek- commented:
The last commit on this project is in 2016.

@webmozart seems not to be maintaining any open source project for a while... Maybe this project will never be updated?!

LIQRGV commented:
I was thinking to move it to my public repo @Nek-.
Do you think it's good idea ?

Nek- commented:
@LIQRGV shouldn't this move to friendsofphp instead?

Nek- commented:
cc @lyrixx (context: https://github.com/webmozart/glob/issues/23 )

LIQRGV commented:
> @LIQRGV shouldn't this move to friendsofphp instead?

I'm ok anywhere as long as there is someone who maintain it

duxthefux commented:
@LIQRGV what's the status here? as help needed somewhere?

LIQRGV commented:
Hi @duxthefux and @Nek-,
what should I do to add this to friendsofphp repo?

Nek- commented:
I think we should wait for @webmozart because he used to give rights to projects. Maybe somebody can reach him (I know he's busy right now with important real-life stuff but we really need it) so he gives you access to this repository.

martinssipenko commented:
I reached out to him via email and got this response:

```
I would be glad if someone took over the lead of the package. I've seen the PRs, but I'm lacking the time to work on them. Would you like to take over?
```

So the question is, where could this project be moved to? Are FriendsOfPHP willing to take over (cc @lyrixx), perhaps Psalm (cc @muglug) wants to take over as they are the largest project that uses this

Nek- commented:
@martinssipenko I think keeping the project here is fine. We just need to elect a maintainer. @LIQRGV seems to be a good candidate WDYT? ^^

_(if nobody wants to do it I can do it)_

LIQRGV commented:
I'm ok with anything as long as this project updated.
😁

MidnightDesign commented:
No offense to @LIQRGV, but @Nek- _seems_ a little more experienced with open source. Maybe both, so we don't end up in the same situation again someday. : )

martinssipenko commented:
I imagine @Ocramius has a ton of stuff on his plate but perhaps he wants to take over? He does have the most experience with OS.

Nek- commented:
@martinssipenko if you can ask for @LIQRGV and me to have access to the repository it would be fine. We'll try to ensure it works properly in the future. (we need to end this fast instead of useless debate ;) )

Ocramius commented:
@martinssipenko I don't mind helping, but the biggest problem is usually getting someone to fiddle with the repo configuration, which doesn't work for contributors.

I'd push some release automation tool ASAP, so that we can ease merges, but need access to repo secrets to do so.

Push rights are the smaller part of the task here 😅

martinssipenko commented:
> @martinssipenko I don't mind helping, but the biggest problem is usually getting someone to fiddle with the repo configuration, which doesn't work for contributors.
> 
> I'd push some release automation tool ASAP, so that we can ease merges, but need access to repo secrets to do so.
> 
> Push rights are the smaller part of the task here 😅

So perhaps transferring/cloning to another account is better approach?

Ocramius commented:
> So perhaps transferring/cloning to another account is better approach?

Absolutely: it allows multiple owners, which is kinda needed for redundancy.

martinssipenko commented:
Frankly, I don't know what to do next, reach out to friendsofphp?

LIQRGV commented:
Transferring to another repo would lead to updating any repo that use this into new repo.
Just saying

localheinz commented:
@LIQRGV 

> Transferring to another repo would lead to updating any repo that use this into new repo.

Not necessary:

- we could continue to use `webmozart/glob` as package name
- we could use a different package name and replace `webmozart/glob` (while keeping the same namespaces)

martinssipenko commented:
That is true, but not transferring means things don't change as the original maintainer has no time to fiddle with repo config. If transferred this could marked as abandoned and suggest using a fork.

@Ocramius you wanna fork?

duxthefux commented:
If we're forking, this would basically mean this repo needs be archived and a replacement suggestions needs to be placed in README and  `composer.json` to make a good dev experience.

Nek- commented:
If an organization should fork this repo, it should be psalm. This is the project that needs it the most. (5 573 473 downloads for this one and 4 871 588 for psaml)

martinssipenko commented:
@Nek- I took a closer look, Psalm stopped using this since version 4.1.0.

webmozart commented:
Hi all! Sorry for being so unresponsive. I'll do my best to help you fix this quickly. Let me consider a few options and then get back to you with proposals.

martinssipenko commented:
This repo has now been moved to `webmozarts` org and @Ocramius and @Nek- have been made as repository maintainers.

Ocramius commented:
Ported @LIQRGV's commits to #24 - waiting for some administrative mumbo-jumbo to happen before I can start rolling out releases :-)
